lms/djangoapps/verify_student/models.py

```diff
@@ -1046,7 +1051,9 @@ def add_status_from_checkpoints(cls, checkpoints, user, status):
             None
         """
         for checkpoint in checkpoints:
-            cls.objects.create(checkpoint=checkpoint, user=user, status=status)
+            # get 'location_id' from last entry and add it in new entry
+            location_id = cls.objects.filter(checkpoint=checkpoint)[0].location_id
```

Comment:
I'd prefer to make this check a little more defensive.  Could you try to retrieve the object, but leave location_id blank if it can't be found?

We need to be really, really careful with this method.  If an exception gets raised here, it could cause the view-level transaction to get rolled back, which means we'd lose the student's verification status.